lib/api/v1/repositories.rb

```diff
@@ -105,6 +105,24 @@ class Repositories < Grape::API
             authorize repo, :show?
             present repo, with: API::Entities::Repositories, type: current_type
           end
+
+          desc "Delete repository",
+               params:  API::Entities::Repositories.documentation.slice(:id),
+               failure: [
+                 [401, "Authentication fails"],
+                 [403, "Authorization fails"],
+                 [404, "Not found"]
+               ]
+
+          delete do
+            repository = Repository.find(params[:id])
+            authorize repository, :destroy?
+
+            destroy_service = ::Repositories::DestroyService.new(current_user)
+            destroyed = destroy_service.execute(repository)
+
+            error!({ "errors" => destroy_service.error }, 422, header) unless destroyed
```

Comment:
On success a delete request should return a 204 status code.

Reply:
Remember that in the other PR about managing team member (still pending btw, but I need to re-review and rebase before asking another review) I was having an issue by writing just the unless? For some reason this is working now. Maybe gem version has been updated? Response seems fine as you can see below:

```
Request URL:http://localhost:3000/api/v1/repositories/12
Request Method:DELETE
Status Code:204 No Content
```

Reply:
Cool. As long as the tests works I'm fine then :+1: